trillian/migrillian/core/controller.go

```diff
@@ -210,24 +207,39 @@ func (c *Controller) RunWhenMaster(ctx context.Context) error {
 	}
 }
 
+// runWithRestarts calls Run until the context is done, in continuous mode. For
+// non-continuous mode it is simply equivalent to Run.
+func (c *Controller) runWithRestarts(ctx context.Context) error {
+	if !c.opts.Continuous {
+		return c.Run(ctx)
+	}
+	for ctx.Err() == nil {
+		if err := c.Run(ctx); err != nil {
```

Comment:
Thinking about whether to restart if `Run` returns nil. Probably I should return in this case, as Run stops gracefully when it reaches max running time, and that is the time when we should resign.